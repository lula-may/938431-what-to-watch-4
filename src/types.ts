export interface Review {
  author: string;
  authorId: number;
  date: Date;
  id: number;
  rating: number;
  text: string;
}

export interface Movie {
  actors: string[];
  bgColor: string;
  bgPoster: string;
  src: string;
  description: string;
  director: string;
  genre: string;
  id: number;
  isFavorite: boolean;
  poster: string;
  previewPoster: string;
  rating: {
    count: number,
    score: number,
  };
  releaseYear: number;
  runTime: number;
  title: string;
};
