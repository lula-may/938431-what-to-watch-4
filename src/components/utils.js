import {DEFAULT_GENRE} from "../const.js";

const getMoviesByGenre = (genre, allMovies) => {
  if (genre === DEFAULT_GENRE) {
    return allMovies;
  }
  return allMovies.filter((movie) => movie.genre === genre);
};

export {getMoviesByGenre};
