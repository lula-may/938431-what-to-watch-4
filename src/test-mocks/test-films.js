const movieTitles = [
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const movies = movieTitles.map((title, i) => {
  return {
    title,
    id: i,
    src: `poster-${i}.jpg`
  };
});

const headerMovie = {
  title: `Jurassic Park`,
  genre: `Sci-Fi`,
  releaseYear: 1993,
  bg: `background.jpg`,
  poster: `header-poster.jpg`
};

export {movies, headerMovie};

