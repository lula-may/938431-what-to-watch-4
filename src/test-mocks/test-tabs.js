const tabs = [
  {
    id: `aa`,
    type: `First`
  },
  {
    id: `bb`,
    type: `Second`,
  },
  {
    id: `cc`,
    type: `Third`,
  }
];

export {tabs};
