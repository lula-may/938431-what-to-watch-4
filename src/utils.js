const HOUR = 60;
const DATE_STRING_LENGTH = 10;
const SECONDS_IN_HOUR = 3600;
const SECONDS_IN_MINUTE = 60;

const extend = (a, b) => Object.assign({}, a, b);

const formatTime = (duration) => {
  const hours = Math.floor(duration / HOUR);
  const minutes = duration % HOUR;

  return (
    (hours > 0)
      ? `${hours}h ${minutes}m`
      : `${minutes}m`
  );
};

const getTwoDigitNumber = (number) => number < 10 && `0${number}` || `${number}`;

const formatElapsedTime = (durationInSeconds) => {
  const duration = Math.round(durationInSeconds);
  const hours = Math.floor(duration / SECONDS_IN_HOUR);
  const minutes = Math.floor(duration % SECONDS_IN_HOUR / SECONDS_IN_MINUTE);
  const seconds = duration % SECONDS_IN_MINUTE;
  return `${getTwoDigitNumber(hours)}:${getTwoDigitNumber(minutes)}:${getTwoDigitNumber(seconds)}`;
};

const formatDate = (date) => date.toLocaleString(`en-US`, {day: `numeric`, month: `long`, year: `numeric`});

const formatDateTimeAttribute = (date) => date.toISOString().slice(0, DATE_STRING_LENGTH);

export {extend, formatDate, formatTime, formatDateTimeAttribute, formatElapsedTime};
