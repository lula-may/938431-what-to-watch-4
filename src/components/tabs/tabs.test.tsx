import * as React from "react";
import * as renderer from "react-test-renderer";

import {noop} from "../../utils";
import Tabs from "./tabs";
import {TabType} from "../../types";
import {reviews, testMovies} from "../../test-mocks/test-films";

const movie = testMovies[0];

describe(`Tabs Component`, () => {
  it(`should render Overview component inside`, () => {
    const tree = renderer.create(
        <Tabs
          activeItem={TabType.OVERVIEW}
          comments={reviews}
          hasLoadingError={false}
          movie={movie}
          onClick={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render Details component inside`, () => {
    const tree = renderer.create(
        <Tabs
          activeItem={TabType.DETAILS}
          comments={reviews}
          hasLoadingError={false}
          movie={movie}
          onClick={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render Reviews Component inside`, () => {
    const tree = renderer.create(
        <Tabs
          activeItem={TabType.REVIEWS}
          comments={reviews}
          hasLoadingError={false}
          movie={movie}
          onClick={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
