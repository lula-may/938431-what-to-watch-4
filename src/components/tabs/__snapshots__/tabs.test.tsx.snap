// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component should render Details component inside 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Overview"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Details"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Reviews"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Stiven Spielberg
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Sam Neill,
          <br />
          Jeff GoldBlum,
          <br />
          Laura Dern
          <br />
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          1h 0m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Sci-Fi
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2000
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Tabs Component should render Overview component inside 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Overview"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Details"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Reviews"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      9.7
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very Good
      </span>
      <span
        className="movie-rating__count"
      >
        150
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      bla-bla-bla, olya-lya-lya, wow-wow
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Stiven Spielberg
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Sam Neill, Jeff GoldBlum, Laura Dern and others
      </strong>
    </p>
  </div>
</div>
`;

exports[`Tabs Component should render Reviews Component inside 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Overview"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Details"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          id="Reviews"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Should old acquaintance be forgot and never brought to mind.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Robert Burns
            </cite>
            <time
              className="review__date"
              dateTime="2020-03-02"
            >
              March 2, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8.8
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Jack and Jill went up the hill.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Jack Daniels
            </cite>
            <time
              className="review__date"
              dateTime="2020-06-02"
            >
              June 2, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          7.7
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Betty Boughter bought some butter.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Betty Boughter
            </cite>
            <time
              className="review__date"
              dateTime="2019-07-15"
            >
              July 15, 2019
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          6.6
        </div>
      </div>
    </div>
  </div>
</div>
`;
